test/fake_code_signer/pe_wrapper.rb

```diff
@@ -0,0 +1,123 @@
+class FakeCodeSigner
+  # This class exists to navigate and manipulate the data of the Windows
+  # executable format, PE, see: https://en.wikipedia.org/wiki/Portable_Executable
+  # also https://msdn.microsoft.com/en-us/library/ms809762.aspx
+  class PEWrapper
+    # size of a DWORD is 2 words (4 bytes)
+    DWORD_SIZE = 4
+
+    # struct IMAGE_DOS_HEADER
+    # https://www.nirsoft.net/kernel_struct/vista/IMAGE_DOS_HEADER.html
+    DOS_HEADER_SIZE = 64
+
+    # first field from IMAGE_NT_HEADERS
+    # https://msdn.microsoft.com/en-us/library/windows/desktop/ms680336(v=vs.85).aspx
+    PE_SIGNATURE_SIZE = 4
+
+    # struct IMAGE_FILE_HEADER
+    # see: https://msdn.microsoft.com/en-us/library/windows/desktop/ms680313(v=vs.85).aspx
+    IMAGE_FILE_HEADER_SIZE = 20
+
+    # struct IMAGE_OPTIONAL_HEADER
+    # https://msdn.microsoft.com/en-us/library/windows/desktop/ms680339(v=vs.85).aspx
+    IMAGE_OPTIONAL_HEADER_SIZE = 224
+
+    # http://bytepointer.com/resources/pietrek_in_depth_look_into_pe_format_pt1_figures.htm
+    NUMBER_OF_DATA_DIRECTORY_ENTRIES = 16
+
+    # struct IMAGE_DATA_DIRECTORY
+    # https://msdn.microsoft.com/en-us/library/windows/desktop/ms680305(v=vs.85).aspx
+    DATA_DIRECTORY_ENTRY_SIZE = 8
+
+    def initialize(image)
+      @image = image
+    end
+
+    # set digital signature address in security header
+    def security_address=(offset)
+      @image[security_address_offset, DWORD_SIZE] = raw_bytes(offset)
```

Comment:
Wasn't familiar with this king of substitution. TIL

Reply:
@jakedouglas yea, it's very handy for this kind of situation!